src/request.ts

```diff
@@ -37,23 +36,57 @@ export async function request<Query, Variables>(
 
     const url = ctx.apiUrl || defaultUrl;
 
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    const { data: res, headers: responseHeaders } = await axios.post(
-      url,
-      {
+    const result = await fetch(url, {
```

Comment:
Do we need to care about try/catching this?

Reply:
Yes so we can _always_ return an error vs throwing. Makes typescript types/handling easier imo.